### ISSUE TITLE
OSError was done at Standalone mode in DL-Framework Windows chainer

### ISSUE THREAD

**Issue by yoshihingis** [Nuitka/Nuitka#417]
Now I have been checking that nuitka is useful for deep-learning framework.
So I check  nuitka  at "Chainer".
But I faced on OSError:exception :access violation reading  0x000000004CBFA4B4
![chainer](https://user-images.githubusercontent.com/18341725/58844066-782b9900-86b0-11e9-836c-9457e9e7139c.jpg)

・Enviroment
  Windows10 Pro 64bit
  CUDA10
  cuDNN Version7.6.0
  Microsoft Visual Studio2015
  Nuitka 0.6.3.1
  Python 3.6.1

・the Way of installing chainer
   (1)pip install cupy-cuda100
   (2)pip install chainer
 
・Python train_mnist is OK
![python_train_mnist](https://user-images.githubusercontent.com/18341725/58844275-6e566580-86b1-11e9-8438-ee91b01ad2b6.jpg)

・train_mnist.exe was OK ,it was compiled by python -m nuitka --recurse-all train_mnist.py
![normal_exe_mnist](https://user-images.githubusercontent.com/18341725/58844333-b5dcf180-86b1-11e9-9249-05a809645378.jpg)

・But in the Standalone mode ,nuitka was errored.
![chainer](https://user-images.githubusercontent.com/18341725/58844562-ae6a1800-86b2-11e9-8abd-3ee1471a8f08.jpg)

 ※Could you suggest any idea to solve this issue.

・Chainer train_mnist.py   
   you can download this example at
https://github.com/chainer/chainer/tree/master/examples/mnist

#!/usr/bin/env python
import argparse

import chainer
import chainer.functions as F
import chainer.links as L
from chainer import training
from chainer.training import extensions
import chainerx

import matplotlib
matplotlib.use('Agg')

# Network definition
class MLP(chainer.Chain):

    def __init__(self, n_units, n_out):
        super(MLP, self).__init__()
        with self.init_scope():
            # the size of the inputs to each layer will be inferred
            self.l1 = L.Linear(None, n_units)  # n_in -> n_units
            self.l2 = L.Linear(None, n_units)  # n_units -> n_units
            self.l3 = L.Linear(None, n_out)  # n_units -> n_out

    def forward(self, x):
        h1 = F.relu(self.l1(x))
        h2 = F.relu(self.l2(h1))
        return self.l3(h2)

def main():
    parser = argparse.ArgumentParser(description='Chainer example: MNIST')
    parser.add_argument('--batchsize', '-b', type=int, default=100,
                        help='Number of images in each mini-batch')
    parser.add_argument('--epoch', '-e', type=int, default=20,
                        help='Number of sweeps over the dataset to train')
    parser.add_argument('--frequency', '-f', type=int, default=-1,
                        help='Frequency of taking a snapshot')
    parser.add_argument('--device', '-d', type=str, default='-1',
                        help='Device specifier. Either ChainerX device '
                        'specifier or an integer. If non-negative integer, '
                        'CuPy arrays with specified device id are used. If '
                        'negative integer, NumPy arrays are used')
    parser.add_argument('--out', '-o', default='result',
                        help='Directory to output the result')
    parser.add_argument('--resume', '-r', type=str,
                        help='Resume the training from snapshot')
    parser.add_argument('--unit', '-u', type=int, default=1000,
                        help='Number of units')
    group = parser.add_argument_group('deprecated arguments')
    group.add_argument('--gpu', '-g', dest='device',
                       type=int, nargs='?', const=0,
                       help='GPU ID (negative value indicates CPU)')
    args = parser.parse_args()

    device = chainer.get_device(args.device)

    print('Device: {}'.format(device))
    print('# unit: {}'.format(args.unit))
    print('# Minibatch-size: {}'.format(args.batchsize))
    print('# epoch: {}'.format(args.epoch))
    print('')

    # Set up a neural network to train
    # Classifier reports softmax cross entropy loss and accuracy at every
    # iteration, which will be used by the PrintReport extension below.
    model = L.Classifier(MLP(args.unit, 10))
    model.to_device(device)
    device.use()

    # Setup an optimizer
    optimizer = chainer.optimizers.Adam()
    optimizer.setup(model)

    # Load the MNIST dataset
    train, test = chainer.datasets.get_mnist()

    train_iter = chainer.iterators.SerialIterator(train, args.batchsize)
    test_iter = chainer.iterators.SerialIterator(test, args.batchsize,
                                                 repeat=False, shuffle=False)

    # Set up a trainer
    updater = training.updaters.StandardUpdater(
        train_iter, optimizer, device=device)
    trainer = training.Trainer(updater, (args.epoch, 'epoch'), out=args.out)

    # Evaluate the model with the test dataset for each epoch
    trainer.extend(extensions.Evaluator(test_iter, model, device=device))

    # Dump a computational graph from 'loss' variable at the first iteration
    # The "main" refers to the target link of the "main" optimizer.
    # TODO(niboshi): Temporarily disabled for chainerx. Fix it.
    if device.xp is not chainerx:
        trainer.extend(extensions.DumpGraph('main/loss'))

    # Take a snapshot for each specified epoch
    frequency = args.epoch if args.frequency == -1 else max(1, args.frequency)
    trainer.extend(extensions.snapshot(), trigger=(frequency, 'epoch'))

    # Write a log of evaluation statistics for each epoch
    trainer.extend(extensions.LogReport())

    # Save two plot images to the result dir
    trainer.extend(
        extensions.PlotReport(['main/loss', 'validation/main/loss'],
                              'epoch', file_name='loss.png'))
    trainer.extend(
        extensions.PlotReport(
            ['main/accuracy', 'validation/main/accuracy'],
            'epoch', file_name='accuracy.png'))

    # Print selected entries of the log to stdout
    # Here "main" refers to the target link of the "main" optimizer again, and
    # "validation" refers to the default name of the Evaluator extension.
    # Entries other than 'epoch' are reported by the Classifier link, called by
    # either the updater or the evaluator.
    trainer.extend(extensions.PrintReport(
        ['epoch', 'main/loss', 'validation/main/loss',
         'main/accuracy', 'validation/main/accuracy', 'elapsed_time']))

    # Print a progress bar to stdout
    trainer.extend(extensions.ProgressBar())

    if args.resume is not None:
        # Resume from a snapshot
        chainer.serializers.load_npz(args.resume, trainer)

    # Run the training
    trainer.run()

if __name__ == '__main__':
    main()

**Comment by kayhayen** [comment]
This one might be better on develop branch already, did you try it? I am about to release it as 0.6.4 and would appreciate knowing that.

**Comment by yoshihingis** [comment]
Thank you for your advice.
I used nuitka 0,6,4rc6 an python3.7.3.
Win10 64bit,CUDA10,VS2015,cuDNN7.6.0
![nuitka064rc6](https://user-images.githubusercontent.com/18341725/59029085-4e34cb00-8898-11e9-8fd4-6073688ebd80.jpg)

(1) I delete chainer ,numpy ,cupy by pip uinstall and reinstall again.
    python chainer train_mnist.py operatted correctly.
![pyhton_chainer](https://user-images.githubusercontent.com/18341725/59029232-aff53500-8898-11e9-9180-054873b41f56.jpg)

(2)I did standalone mode compile
![standalone_comple](https://user-images.githubusercontent.com/18341725/59030049-d2884d80-889a-11e9-9dbf-6119a4fc74f3.jpg)

 (3)I checkers folder/train_mnist.dist and I could find train_mnist.exe
![dist](https://user-images.githubusercontent.com/18341725/59030141-18ddac80-889b-11e9-94eb-3e50178887c8.jpg)

(4)But train_mnist.exe -g 0 did not operate correctly
![exe_error](https://user-images.githubusercontent.com/18341725/59030189-3f034c80-889b-11e9-9f76-c3aa211f26da.jpg)

(5)
 ・I installed pyhton3.7.3 only ,I did not use anaconda, pyenv and etc,
 ・I installed cupy 　＄python3 -m pip install cupy-cuda100
 ・I installer numpy  　＄python3 -m pipi nstall numpy.
    I uinstalled numpy and re-install again, train_mnist.exe didi not operate correctly.

(6)Do you have any idea to solve this issue?

Regards,

**Comment by yoshihingis** [comment]
(1) I changed Nuitka complie options as below.
     During Nuitka was compiling ,there were not any warnings and erros 
 
![chainer06101](https://user-images.githubusercontent.com/18341725/59172654-399c4f80-8b84-11e9-83c3-2ebe0c6f445d.jpg)

python -m nuitka --standalone --plugin-enable=numpy 
--plugin-enable=pylint-warnings 
--include-plugin-directory=C:\Users\e10853\AppData\Local\Programs\Python\Python37\lib\site-packages\numpy train_mnist.py

(2) "train_mnist.exe" was in the folder train_mnist.dist , I run it 
![chainer06102](https://user-images.githubusercontent.com/18341725/59172701-77997380-8b84-11e9-8835-4deceaf73c21.jpg)

(3) Error messages were changed.

(4) I'd really appreciate it if you will give your opnion !

Regards,

**Comment by kayhayen** [comment]
Can I ask you to open a new issue for this new finding. I am closing this. The other bug is more the domain of others in our team.
